### ISSUE TITLE
Run Stage 4b Canary B after v3 NaN fix (link label masking)

### ISSUE THREAD

**Issue by AliAzizpourian3** [AliAzizpourian3/kvp10k-thesis#1]
## Task for Copilot

Pull the latest `main` branch and rerun the Stage 4b Canary B training job on the cluster.

### What changed (commit `95f10dd`)
- `code/script/layoutlm_model.py` — v3 NaN fix: `valid_mask` guard added before BCE in the link loss block. This was the **true root cause** of all NaN crashes. Invalid float values in `gt_links` (outside `[0,1]`) were being silently passed to `binary_cross_entropy_with_logits`, producing `link_loss=NaN` at step 2488 in Canary B.
- `STAGE4_METHODOLOGY_DRAFT.tex` — Updated with full NaN root cause trail (DRAFT).

### Steps to run

```bash
# On the cluster login node:
cd /home/woody/iwi5/iwi5413h/kvp10k_thesis
git pull origin main

# Resubmit Canary B (lr=2e-5, accum=8)
sbatch slurm/kvp_stage4b_canary_B.sbatch
```

### Expected outcome
- Training should proceed past step 2488 without NaN
- `link_loss` should remain finite throughout all epochs
- If stable, proceed to submit full lambda sweep: λ ∈ {0.5, 1.0, 2.0}

### Paths (do NOT change these)
- Repo root: `/home/woody/iwi5/iwi5413h/kvp10k_thesis`
- Python: `/home/woody/iwi5/iwi5413h/kvp10k_thesis/kvp10k_env/bin/python`
- Scripts: `/home/woody/iwi5/iwi5413h/kvp10k_thesis/code/script/`
- Data: `/home/woody/iwi5/iwi5413h/kvp10k_thesis/data/prepared`
- Outputs: `/home/woody/iwi5/iwi5413h/kvp10k_thesis/data/outputs/`
